server/main.go

```diff
@@ -47,25 +54,29 @@ func main() {
 	r.Run(":" + strconv.Itoa(config.Port))
 }
 
-func NewApp() App {
+// NewApp creates the app that holds all the functions to interact with the DB
+
+func NewApp(config *datastore.DatabaseConfig) App {
 	app := App{
-		db: &inmemory.InMemory{},
+		db: &psql.DB{},
 	}
-	err := app.db.Init()
+	err := app.db.Init(config)
+	fmt.Println(err)
 	if err != nil {
 		log.Fatal("could not initialize database")
 	}
-
 	return app
 }
 
-func ReadConfig(file string) (*ServerConfig, error) {
```

Comment:
Keep this to use ServerConfig though, because with the tag `yaml:"database"` on the `DBConfig` field the `Unmarshal` method later on should just work to convert the config file into the `ServerConfig` struct, including the `DBConfig` field

Reply:
Yeah, just tested it and it works like a charm. Just make sure that you use the `ServerConfig`'s `DBConfig` field to initialize the db (in `NewApp`)